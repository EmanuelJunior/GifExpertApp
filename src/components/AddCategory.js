import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // String

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = ( event ) => {
        event.preventDefault();

        if( inputValue.trim().length >= 2 ){
            setCategories( categories => [inputValue, ...categories] );
            setInputValue('');
            console.log('Llegamos hasta aca!');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={ inputValue }
                // onChange es parecido a el evento del teclado keydown
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired
}

