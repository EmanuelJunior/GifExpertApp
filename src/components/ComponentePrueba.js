import { useEffect } from 'react';
import { Fragment } from 'react';

const ComponentePrueba = () => {

    const saludar = () => alert("Hello world");
    saludar();

    return (
        <Fragment>
            <h1 className="titulo" id='titulo' contentEditable={true}>Hola mundo desde Colom</h1>
            <p id="parr1">Esto es el primer parrafo</p>
            <p id="parr2">Este es el segundo parrafo</p>
            <b>Hola estoy en bold</b>

            <div className='anidado'>
                <p>Un parrafo</p>
                <h3>Un subtitulo</h3>
                <hr></hr>
                
                <input 
                    type='text' 
                    placeholder='Users'
                    className='userInput' 
                    value='Aca debe de ir el usuario'
                />
            </div>
        </Fragment>
    );
}

export default ComponentePrueba;