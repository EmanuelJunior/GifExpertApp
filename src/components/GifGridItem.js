import React from 'react'

export const GifGridItem = ({ id, title, url }) => {
    
    return (
        <div className='card animate__animated animate__bounceInRight'>
            <img src={ url } alt={ title }/>
            <p> { title } </p>
        </div>
    )
}
