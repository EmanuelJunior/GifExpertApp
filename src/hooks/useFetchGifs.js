import { useEffect, useState } from "react";
import { getGifs } from '../helpers/getGifs';

/* los custom hooks funcionan como funcional components 
peden tener las mismas funcionalidades */

export const useFetchGifs = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect( ()=> {

        getGifs( category )
            .then( imgs => {
                    
                setState({
                    data: imgs,
                    loading: false
                });

            })
    }, [category]);

    return state; // { data: [ imgs ], loading: true };
}