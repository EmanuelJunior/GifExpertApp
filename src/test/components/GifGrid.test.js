import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el <GifGrid />', () => {

    const category = "One Punch Man";
    
    test('Debe de mostrarse correctamente', () => {
 
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        expect( wrapper ).toMatchSnapshot();
 
    });

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });    
})
