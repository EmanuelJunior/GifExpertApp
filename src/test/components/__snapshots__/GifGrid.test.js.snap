// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
     
    One Punch Man
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
     
    One Punch Man
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
